state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1141,17 +1143,41 @@ pub mod pallet {
 		}
 
 		/// Associates `short_id` with `affiliate_id` for a given broker. Overwrites the record
-		/// under `short_id` if already taken by another affiliate.
+		/// under `short_id` if already taken by another affiliate. Leave `short_id` empty to auto
+		/// assign one.
 		#[pallet::call_index(14)]
 		#[pallet::weight(T::WeightInfo::register_affiliate())]
 		pub fn register_affiliate(
 			origin: OriginFor<T>,
-			short_id: AffiliateShortId,
 			affiliate_id: T::AccountId,
+			short_id: Option<AffiliateShortId>,
```

Comment:
I don't think we should change the extrinsic. 

My suggestion was to do the checks before calling the extrinsic.